AudioDeviceExample/AudioDevices/ExampleAVAudioEngineDevice.m

```diff
@@ -797,6 +797,10 @@ static OSStatus ExampleAVAudioEngineDeviceRecordCallback(void *refCon,
     }
 
     AudioCapturerContext *context = (AudioCapturerContext *)refCon;
+    
+    if (context->deviceContext == NULL) {
+        return noErr;
+    }
```

Comment:
It seems like the right solution is to manage the lifecycle of the audio device so that this scenario does not occur. But I guess the workaround is needed because the audio unit is statically configured. As an incremental step I think it is a good one, but I wouldn't stop here.